app/controllers/tasks/collection_objects/area_and_date_controller.rb

```diff
@@ -0,0 +1,137 @@
+    set_and_order_dates(params)
+
+    if @shape_in.blank? and @geographic_area_id.blank? # missing "? " was fixed
+      area_object_ids = CollectionObject.where('false')
+    else
+      area_object_ids = GeographicItem.gather_selected_data(@geographic_area_id, @shape_in, 'CollectionObject').map(&:id)
+      area_set = true
+    end
+
+    if (@start_date.blank? || @end_date.blank?) #|| area_object_ids.count == 0
+      @collection_objects = CollectionObject.where('false')
+    else
+      collecting_event_ids = CollectingEvent.in_date_range(date_range_params).pluck(:id)
+      @collection_objects  = CollectionObject.from_collecting_events(collecting_event_ids,
+                                                                     area_object_ids,
+                                                                     area_set,
+                                                                     sessions_current_project_id).page(params[:page])
+    end
+
+    @collection_objects_count = @collection_objects.count
+    @feature_collection       = ::Gis::GeoJSON.feature_collection(find_georeferences_for(@collection_objects,
+                                                                                         @geographic_area))
+    message                   = 'No collection objects found.' if @collection_objects_count == 0
+    @package                  = render_co_select_package(message)
+    # render_co_select_json(message)
+  end
+
+  # GET
+  # @return [Object] json object containing count
+  def set_area
+    @geographic_area_id       = params[:geographic_area_id]
+    @shape_in                 = params[:drawn_area_shape]
+    @collection_objects_count = GeographicItem.gather_selected_data(@geographic_area_id,
+                                                                    @shape_in,
+                                                                    'CollectionObject').count
+    render json: {html: @collection_objects_count.to_s}
+  end
+
+  # GET
+  # @return [Object] json object containing count and html of the chart
+  def set_date
+    set_and_order_dates(params)
+    @collection_objects       = CollectionObject.in_date_range(date_range_params)
+    @collection_objects_count = @collection_objects.count
+    chart                     = render_to_string(partial: 'stats',
+                                                 locals:  {count:   @collection_objects_count,
+                                                           objects: @collection_objects})
+    render json: {html: @collection_objects_count.to_s, chart: chart}
+  end
+
+  # GET
+  def set_taxon_name
+    @taxon_name_id            = params[:taxon_name_id]
+    @taxon_name               = TaxonName.find(@taxon_name_id)
+    @collection_objects       = CollectionObject.where('false')
+    @collection_objects_count = @collection_objects.count
+    render json: {html: @collection_objects_count.to_s}
+  end
+
+  def download_result
```

Comment:
Perhaps comment with the GET line that this is intended for, is is js or HTML?  Also for other empty defs.